library/public/options.h.in

```diff
@@ -0,0 +1,138 @@
+#ifndef f3d_options_h
+#define f3d_options_h
+
+#include "exception.h"
+#include "export.h"
+#include "types.h"
+
+#include <array>
+#include <string>
+#include <vector>
+
+#ifndef F3D_DISABLE_CXX17_API
+  #include <variant>
+#endif
+
+namespace f3d
+{
+#ifndef F3D_DISABLE_CXX17_API
+  // Declared here for simplicity
+  using option_variant_t = std::variant<bool, int, double, std::string, std::vector<double>>;
+#endif
```

Comment:
I think clang-format does not catch it's generated but there's no identation after a `#ifdef`
```suggestion
#ifndef F3D_DISABLE_CXX17_API
// Declared here for simplicity
using option_variant_t = std::variant<bool, int, double, std::string, std::vector<double>>;
#endif
```

Reply:
indeed, I should add that to clang format though.